packages/payload/src/query-presets/preventLockout.ts

```diff
@@ -0,0 +1,92 @@
+import type { Validate } from '../fields/config/types.js'
+
+import { APIError } from '../errors/APIError.js'
+import { createLocalReq } from '../utilities/createLocalReq.js'
+import { initTransaction } from '../utilities/initTransaction.js'
+import { killTransaction } from '../utilities/killTransaction.js'
+import { queryPresetsCollectionSlug } from './config.js'
+
+/**
+ * Prevents "accidental lockouts" where a user makes an update that removes their own access to the preset.
+ * This is effectively an access control function proxied through a `validate` function.
+ * How it works:
+ *   1. Creates a temporary record with the incoming data
+ *   2. Attempts to read and update that record with the incoming user
+ *   3. If either of those fail, throws an error to the user
+ *   4. Once finished, prevents the temp record from persisting to the database
+ */
+export const preventLockout: Validate = async (
+  value,
+  { data, overrideAccess, req: incomingReq },
+) => {
+  // Use context to ensure an infinite loop doesn't occur
+  if (!incomingReq.context.isValidationReq && !overrideAccess) {
+    const req = await createLocalReq(
+      {
+        context: {
+          isValidationReq: true,
```

Comment:
I would prefix this with `_` to avoid potential conflicts and if anyone sees it they know it is not their own project code. Also we might rename it to match this function name `preventLockout` so we know where it was coming from when we see it while debugging.